modules/rank-eval/src/main/java/org/elasticsearch/index/rankeval/QuerySpec.java

```diff
@@ -32,25 +36,33 @@
  * Defines a QA specification: All end user supplied query intents will be mapped to the search request specified in this search request
  * template and executed against the targetIndex given. Any filters that should be applied in the target system can be specified as well.
  *
- * The resulting document lists can then be compared against what was specified in the set of rated documents as part of a QAQuery.  
+ * The resulting document lists can then be compared against what was specified in the set of rated documents as part of a QAQuery.
  * */
 public class QuerySpec implements Writeable {
 
-    private int specId = 0;
+    private String specId;
     private SearchSourceBuilder testRequest;
     private List<String> indices = new ArrayList<>();
     private List<String> types = new ArrayList<>();
-    
-    public QuerySpec(
-            int specId, SearchSourceBuilder testRequest, List<String> indices, List<String> types) {
+    /** Collection of rated queries for this query QA specification.*/
+    private List<RatedDocument> ratedDocs = new ArrayList<>();
+
+    public QuerySpec() {
+        // ctor that doesn't require all args to be present immediatly is easier to use with ObjectParser
+        // TODO decide if we can require only id as mandatory, set default values for the rest?
+    }
+
+    public QuerySpec(String specId, SearchSourceBuilder testRequest, List<String> indices, List<String> types,
+            List<RatedDocument> ratedDocs) {
         this.specId = specId;
         this.testRequest = testRequest;
         this.indices = indices;
         this.types = types;
+        this.ratedDocs = ratedDocs;
```

Comment:
Moved the rated docs here because from the proposed structure of the rest request I thought they better belong here.